docs/platforms/aws-lambda/nodejs/index.md

```diff
@@ -0,0 +1,465 @@
+```
+
+## `aws.lambda.initialization`
+
+This is a Span within a Trace for AWS Lambda that represents the time spent loading your AWS Lambda function and running any initialization code.
+
+Initialization includes the following:
+
+Extension Init – Initializing all External AWS Lambda Extensions configured on the function (there is a limit of 10 Extensions per function).
+
+Runtime Init – Initializing the AWS Lambda Runtime (e.g, Node.js, Python).
+
+Function Init – Initializing the AWS Lambda Function code.
+
+Initialization only appears for the first event processed by each instance of your function, which is also known as a Cold-Start. It can also appear in advance of function invocations if you have enabled provisioned concurrency.
+
+You will want to optimize Initialization performance as best you can. Poor Initialization performance will directly affect the experience of your users and customers. Additionally, it's important to note that AWS charges you for Initialization time. Unfortunately, no tooling can offer a breakdown of what happens within the Initialization phase of AWS Lambda. Generally, adding multiple Extensions, large code file sizes, and using a slower runtime (e.g., Java) are the biggest culprits when it comes to slow Initialization.
+
+Once initialized, each instance of your function can process thousands of requests without performing another Initialization. However, AWS Lambda function instance containers will shutdown within 5-15 minutes of inactivity. After that, the next event will be a Cold-Start, causing Initialization to run again.
+
+### Tags
+
+These are the Tags attached to this Span:
+
+```javascript
+/* Tags: Standard */
+
+s.span.id: "ja819kasfj10asfh" // This Span's ID
+s.span.parent.id: "e8a4e2a264145041" // This Span's parent Span ID.
+s.span.timestamp: "2022-07-13T20:10:48.794945024Z" // This Span's time.  Same as start_time.
+s.span.start_time: "2022-07-13T20:10:48.794945024Z" // This Span's start time.
+s.span.end_time: "2022-07-13T20:10:48.799375104Z" // This Span's end time.
+
+/* Tags: aws.lambda */
+ 
+aws.lambda.initialization.duration: 600 // Maps to the Cloudwatch Logs Report "Init Duration"
+```
+
+## `aws.lambda.invocation`
+
+This is a Span within a Trace for AWS Lambda. After Initialization, Extensions and the handler of the AWS Lambda function run in the Invocation phase. This phase includes:
+
+Running External Extensions in parallel with the function. These also continue running after the function has completed, enabling Serverless Console to capture diagnostic information and ingest metrics, traces and logs.
+
+Running the wrapper logic for Internal Extensions.
+
+Running the handler for your AWS Lambda.
+
+The Invocation phase is comprised mostly of your handler (i.e. your business logic), and you want to optimize that as best you can because its performance (combined with Initialization performance) will have the biggest impact on the experience for your users and customers.
```

Comment:
We may further tokenize t the invocation duration into:
- Handler duration
- Extension overhead duration
- Runtime duration (`totalDuration - (handlerDuration + extensionDuration)`)

Reply:
In this specification, we've been careful to only focus on the three high-level Lambda lifecycle phases which AWS has made clear (Initialization, Invocation, Shutdown), stick only to AWS terminology, and only pull durations from those phases from an accurate piece of information given to use directly from Cloudwatch.

I don't know what _Handler Duration_, _Runtime Duration_, and _Extension Overhead Duration_ is.  Can you point to AWS docs that use these terms and describe them?  Also, how would we get this information accurately?

Reply:
It's sort of outlined in this graph: https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html#runtimes-lifecycle-invoke

The actual function invocation duration takes just part of the total invocation duration, especially with the extensions on board.

I think it's valuable to report to the user how long the actual function logic takes time, and how much is consumed by the runtime and eventual extensions